app/controllers/invoices/recalculate_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2022, Schweizer Wanderwege. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+class Invoices::RecalculateController < ApplicationController
+
+  respond_to :js
+  before_action :authorize_action
+  helper_method :entry
+
+  def new;end
+
+  private
+
+  def entry
+    @entry ||= build_entry
+  end
+
+  def build_entry
+    invoice = Invoice.new
+    if params.dig(:invoice_list, :invoice).present?
+      invoice.attributes = params.require(:invoice_list).require(:invoice)
+                                 .permit(InvoicesController.permitted_attrs)
+    else
+      invoice.attributes = params.require(:invoice).permit(InvoicesController.permitted_attrs)
+    end
```

Comment:
Pack das doch noch in eine `#permitted_params` Methode

Das `if` solltest du auch glaub zu `params.key?(:invoice_list)` refactoren können


Dann kannst du hier nur noch `invoice.attributes = permitted_params` callen